base/broadcast.jl

```diff
@@ -10,6 +10,11 @@ export broadcast_getindex, broadcast_setindex!
 
 ## Broadcasting utilities ##
 
+# fallback routines for broadcasting with no arguments or with scalars
+# to just produce a scalar result:
+broadcast(f) = f()
+broadcast(f, x::Number...) = f(x...)
```

Comment:
You're right, the deprecation method seems more specific according to the rules discussed in #16489, but maybe the varargs are confusing things here.  @JeffBezanson?